velox/functions/sparksql/aggregates/FirstLastAggregate.cpp

```diff
@@ -152,18 +201,18 @@ class FirstAggregate : public FirstLastAggregateBase<numeric, TData> {
 
   // If we found a valid value, set to accumulator, then skip remaining rows in
   // group.
-  bool updateValue(vector_size_t i, char* group, DecodedVector& decoded) {
+  bool updateValue(vector_size_t i, char* group, const BaseVector* vector) {
     auto accumulator = exec::Aggregate::value<TAccumulator>(group);
     if (accumulator->has_value()) {
       return false;
     }
 
     if constexpr (!numeric) {
-      return updateNonNumeric(i, group, decoded);
+      return updateNonNumeric(i, group, vector);
     } else {
-      if (!decoded.isNullAt(i)) {
+      if (!vector->isNullAt(i)) {
```

Comment:
This check is not right. You need to check DecodedVector::isNullAt since nulls could be added in encodings on top of the base vector.

Reply:
I will modify this as your suggested, but I wonder whether below two if branch code are equals:
```
    rows.applyToSelected([&](vector_size_t i) {
      if (decoded.isNullAt(i)) {
        // ...
      }
      if (decoded.base()->isNullAt(decoded.index(i))) {
        // ...
      }
    });
```

Reply:
@yuetaoxu No, these are not equal. You can see this by using Dict(Flat) vector where dictionary adds nulls.

Create flat vector with no nulls, call wrapInDictionary with some nulls set in the 'nulls' buffer. Then, decode and test.